### ISSUE TITLE
4.0.1 Basic Authorisation token not being sent in headers

### ISSUE THREAD

**Issue by grahambunce** [domaindrivendev/Swashbuckle.AspNetCore#1127]
I know this is a Duplicate of #603, #647, #653, #624 but they are all closed. This is a problem with 3.x of Swashbuckle and the latest 4.0.1, specifically about **basic** authentication - I don't want to confuse this with the other closed tickets that mix up Bearer or OAuth2 as these appear to work fine in the later versions of SwashBuckle.

The authentication header is not sent either via the Explorer UI or via CURL. I've raised this again as in the other tickets people state that they can "basic" to work.. whereas it no longer appears to in 3.x or 4.x

This is using .net core 2.1.3 with Swashbuckle 4.01.

Not sure what I'm doing wrong, as this appears to work for Bearer/OAuth2.

SwaggerGen:

```csharp
services.AddSwaggerGen(c =>
{
	c.OperationFilter<SecurityRequirements>();
	c.AddSecurityDefinition("basic", new BasicAuthScheme
	{
		Type = "basic",
		Description = "basic authentication using a partner id and secret"
	});
	c.AddSecurityRequirement(new Dictionary<string, IEnumerable<string>>
	{
		{ "basic", new string[] { } }
	});
	c.OperationFilter<SecurityRequirements>();
});
```

I've applied an operation filter and can confirm that the operation.Security line is being hit correctly for those method that have an [Authorize] attribute

```csharp
    public class SecurityRequirements : IOperationFilter
    {
        public void Apply(Operation operation, OperationFilterContext context)
        {
            // Policy names map to scopes
            var requiredScopes = context.MethodInfo.DeclaringType.GetCustomAttributes(true)
                .Union(context.MethodInfo.GetCustomAttributes(true))
                .OfType<AuthorizeAttribute>()
                .Select(a => a.Policy);

            if (!requiredScopes.Any())
            {
                return;
            }
            if (!operation.Responses.ContainsKey("401"))
            {
                operation.Responses.Add("401", new Response { Description = "Unauthorized request. This may be because the request to the service has not been properly authenticated" });
            }
            if (!operation.Responses.ContainsKey("403"))
            {
                operation.Responses.Add("403", new Response { Description = "Forbidden request. This may be because the partner credentials are incorrect for the resource requested" });
            }
            if (!operation.Responses.ContainsKey("405"))
            {
                operation.Responses.Add("405", new Response { Description = "Not allowed request. This may be because the partner is not configured for the type of request" });
            }
            operation.Security = new List<IDictionary<string, IEnumerable<string>>>
            {
                new Dictionary<string, IEnumerable<string>>
                {
                    { "Basic", new string[] { } }       //Supports Swagger Basic authentication
                }
            };
        }
    }
```

Startup.cs

```csharp
services
     .AddAuthentication("BasicAuthentication")
     .AddScheme<AuthenticationSchemeOptions, BasicAuthenticationHandler>("BasicAuthentication", null);
```

**Comment by domaindrivendev** [comment]
With the call to `AddSecurityRequirement`, you're configuring a global requirement (i.e. applicable to _all_ operations by default) for the scheme called "basic". This is setup correctly but then you're also configuring an operation filter that _overrides_ this default requirement with a schema called "Basic". The latter is specified with an uppercase name but the scheme names are case sensitive. Update it to match the definition name and you should be unblocked.

**Comment by grahambunce** [comment]
Thanks for this - the suggested change works fine now
